### PR TITLE
Add client

### PR DESCRIPTION
**Pull request by ahuang11**
Closes

### Example

image

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-bitbucket/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [ ] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-bitbucket/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_bitbucket/credentials.py [hunk 1]
```diff
@@ -35,6 +50,14 @@ class BitBucketCredentials(Block):
         default=None,
         description="Identification name unique across entire BitBucket site.",
     )
+    password: Optional[str] = Field(
+        default=None, description="The password to authenticate to BitBucket."
+    )
+    url: Optional[str] = Field(
+        default=None,
+        description="The base url used for the cloud / local client.",
```
Comment (desertaxle) [inline prefect_bitbucket/credentials.py]:
```suggestion
        description="The base URL of your BitBucket instance.",
```

File: prefect_bitbucket/credentials.py [hunk 2]
```diff
@@ -35,6 +50,14 @@ class BitBucketCredentials(Block):
         default=None,
         description="Identification name unique across entire BitBucket site.",
     )
+    password: Optional[str] = Field(
+        default=None, description="The password to authenticate to BitBucket."
+    )
+    url: Optional[str] = Field(
+        default=None,
```
Comment (desertaxle) [inline prefect_bitbucket/credentials.py]: Should this default to the cloud BitBucket URL?

File: prefect_bitbucket/credentials.py [hunk 3]
```diff
@@ -35,6 +50,14 @@ class BitBucketCredentials(Block):
         default=None,
         description="Identification name unique across entire BitBucket site.",
     )
+    password: Optional[str] = Field(
```
Comment (desertaxle) [inline prefect_bitbucket/credentials.py]: If a password is provided, is a token also needed? Should token be optional now also?

File: prefect_bitbucket/credentials.py [hunk 4]
```diff
@@ -48,3 +71,27 @@ def _validate_username(cls, value: str) -> str:
...
+
+        """
+        # ref: https://atlassian-python-api.readthedocs.io/
+        if isinstance(client_type, str):
+            client_type = ClientType(client_type.lower())
+        client_kwargs = dict(
+            url=self.url, username=self.username, password=self.password
+        )
+        if client_type == ClientType.CLOUD:
+            client = Cloud(cloud=True, **client_kwargs)
```
Comment (desertaxle) [inline prefect_bitbucket/credentials.py]: I don't think that `cloud=True` is necessary. It looks like it gets set to `True` in `Cloud.__init__`: https://github.com/atlassian-api/atlassian-python-api/blob/master/atlassian/bitbucket/cloud/__init__.py
```suggestion
            client = Cloud(**client_kwargs)
```

**Review by desertaxle — CHANGES_REQUESTED** [review]

File: prefect_bitbucket/credentials.py [hunk 5]
```diff
@@ -35,6 +50,14 @@ class BitBucketCredentials(Block):
         default=None,
         description="Identification name unique across entire BitBucket site.",
     )
+    password: Optional[SecretStr] = Field(
+        default=None, description="The password to authenticate to BitBucket."
+    )
+    url: Optional[str] = Field(
```
Comment (ahuang11) [inline prefect_bitbucket/credentials.py]:
```suggestion
    url: str = Field(
```

**Review by desertaxle — APPROVED** [review]

**Pull request merged**